src/components/QueryEditor/VisualQueryEditor/utils/utils.ts

```diff
@@ -220,3 +222,52 @@ export function filterColumns(
       tableSchema?.filter((c) => expression?.columns?.includes(toColumnName(c)))
     : tableSchema;
 }
+
+// return columns in use by an expression. If none has been specified, return the first
+// time and number columns from the table definition
+export function defaultTimeSeriesColumns(expression: QueryExpression, tableColumns: AdxColumnSchema[]): string[] {
+  const res: string[] = [];
+  if (expression.where.expressions?.length) {
+    (expression.where.expressions as QueryEditorArrayExpression[]).forEach((exp) => {
```

Comment:
Is this safe? I assume we'll add betterer to the plugins CI sometime soon, so you might as well use type guards now so you don't have to fix it later.

Reply:
It is, the types of the expressions are a bit of a mess at the moment. It's using wrong definitions that are generally bypassed by legacy code. Let me create a follow up issue to refactor this